src/bson.rs

```diff
@@ -838,3 +845,19 @@ impl From<DateTime<Utc>> for UtcDateTime {
         UtcDateTime(x)
     }
 }
+
+/// Represents a BSON regular expression.
+#[derive(Debug, Clone, PartialEq)]
+pub struct RegExp {
```

Comment:
I named this `RegExp` to match the existing name of the BSON variant. That being said, this would be a good opportunity to change it if we prefer something like `Regex`.

Reply:
I would say I prefer `Regex` since it mirrors the `regex` crate's definition, which I imagine we would want to consider interop with in the future.

Reply:
I'm not convinced that we should be doing interop with the `regex` crate (mostly due to having misgivings about regex being a thing that should even exist at the BSON type level), but I also prefer the name `Regex`, so I'll update it.

Reply:
Hmm, I thought we already had an auto-generated ticket for this, but I can't find it now.